### PR TITLE
Rollback to old mini assembler only example

### PR DESCRIPTION
**Pull request by feliam**
Refactor report generator to handle bytecode only contracts (no metadata) again. RFC/WIP

---
This change is [Reviewable](https://reviewable.io/reviews/trailofbits/manticore/889)

### CODE REVIEW

File: manticore/ethereum.py [hunk 1]
```diff
@@ -1423,41 +1423,6 @@ def finalize(self):
         to a STOP or RETURN in the last symbolic transaction).
         """
 
-        # move runnign states to final states list
```
Comment (mossberg) [inline manticore/ethereum.py]: what was this section moved to the bottom?

Comment (feliam) [reply manticore/ethereum.py]: Ugh...
The problem of not having source/metadata was at the reporting of "global_XXXX" stuff. Reordering this just made it fail faster for debugging.

File: manticore/ethereum.py [hunk 2]
```diff
@@ -1549,13 +1549,13 @@ def global_coverage(self, account_address):
...
             world = self.get_world(state_id)
-            if account_address in world.storage:
+            if account_address in world.contract_accounts:
                 break
 
         with self.locked_context('runtime_coverage') as coverage:
             seen = coverage
-        #runtime_bytecode = world.storage[account_address]['code']
-        runtime_bytecode = self.get_metadata(account_address).runtime_bytecode
+        runtime_bytecode = world.get_code(account_address)[:-44]
```
Comment (mossberg) [inline manticore/ethereum.py]: let's not use the magic number -44 here, either add a comment or a variable or something?

File: manticore/ethereum.py [hunk 3]
```diff
@@ -1549,13 +1548,14 @@ def global_coverage(self, account_address):
...
-            if account_address in world.storage:
+            if account_address in world.contract_accounts:
                 break
 
         with self.locked_context('runtime_coverage') as coverage:
             seen = coverage
-        #runtime_bytecode = world.storage[account_address]['code']
-        runtime_bytecode = self.get_metadata(account_address).runtime_bytecode
+
+        #fixme use this https://github.com/trailofbits/manticore/blob/a98902c81adfa59a86b35e116869609ad1774cdc/manticore/ethereum.py#L173-L179
```
Comment (mossberg) [inline manticore/ethereum.py]: wait, SolidityMetadata._without_metadata exists here though!

File: manticore/ethereum.py (hunk 1 above)
Comment (feliam) [reply manticore/ethereum.py]: Update: This section is moved to the bottom so there is at least one state alive from where we can take the runtime bytecode (then assume that is the same in every state) and calculate the coverage over it.

**Pull request merged**